### PR TITLE
Tweaks to PostgreSQL docker-compose service

### LINKED ISSUE #36: Remove dependency on pyscopg2
The python package pyscopg2 is a large requirement that also brings in heavy dependencies (postgresql-server-dev-*). However, in dmrunner it is only used to check whether the postgresql server is running.

Ideally we would like to have a pure Python way of checking the server response, but so far I have not been able to find anything like that, so we would have to look at writing it ourselves.

Alternatively, a lightweight C command exists called `pg_isready` which is part of `libpq`. Unfortunately, installing this on macOS is a pain (see #24). Another option is we could just rely on the report of `docker-compose`, as this is supposed to ensure itself that the server is up and running.

### PR DESCRIPTION
**Pull request by lfdebrux**
This PR includes two small tweaks:

  1. remove the custom postgres user, DMRUNNER_USER
  2. update the postgres image

On their own these changes aren't very significant, but they lay the ground work needed to fix #36, and also are needed for work to make it possible to use dmrunner on Linux.

### CODE REVIEW

**Comment by samuelhwilliams** [comment]
P.S. I'm also super against commits that are just 'run codestyle', because they break history up a bit. If you're happy going back and running the formatter against each individual commit, so that they bring in the changes in the correct format, that would be fab.

I should figure out a way to enforce this.

File: config/sql/populate_db.sh [hunk 1]
```diff
@@ -1,11 +1,8 @@
...
 
-CREATE_USER_CMD="create user "
-CREATE_USER_CMD+=$DMRUNNER_USER
-CREATE_USER_CMD+=" superuser;"
-
-psql -d postgresql://postgres@localhost:5432 -c "$CREATE_USER_CMD"
-psql -d postgresql://localhost:5432 -c 'create database digitalmarketplace;'
-psql -d postgresql://localhost:5432/digitalmarketplace -c 'create database digitalmarketplace_test;'
-psql -d postgresql://localhost:5432/digitalmarketplace -f /docker-entrypoint-initdb.d/data/*.sql
+psql --username="$POSTGRES_USER" --dbname="$POSTGRES_DB" <<-EOSQL
```
[the remark this answers is not in the document]
Comment (lfdebrux) [reply config/sql/populate_db.sh]: These are generated automatically by the `postgres` image entrypoint, the format that I used for `populate_db.sh` is based on the example in [their documentation](https://hub.docker.com/_/postgres/)

**Comment by lfdebrux** [comment]
> P.S. I'm also super against commits that are just 'run codestyle', because they break history up a bit. If you're happy going back and running the formatter against each individual commit, so that they bring in the changes in the correct format, that would be fab.

That's fair, TBH I pushed this PR out in a rush because it was past my bed time...

File: config/sql/populate_db.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (samuelhwilliams) [reply config/sql/populate_db.sh]: Noice 👍

**Pull request merged**